atest/testdata/standard_libraries/builtin/run_keyword_with_errors.robot

```diff
@@ -134,6 +140,13 @@ Expect Error When Syntax Error In For Loop
     [Documentation]    FAIL  Variable '\${nonexisting}' not found.
     Run Keyword And Expect Error    *    For Loop With Syntax Error
 
+Expect Error When Access To Nonexisting Variable
+    Run Keyword And Expect Error    Variable '\${nonexisting}' not found.    Access To Nonexisting Variable
+
+Expect Error When Access To Nonexisting Extended Variable
+    ${expected_err_msg} =  Set Variable  Resolving variable '\${list[2]}' failed: IndexError: list index out of range
+    Run Keyword And Expect Error  ${expected_err_msg}  Access To Nonexisting Extended Variable
```

Comment:
It's a bit risky to tests the exact error message from Python because it can be different on different Python interpreters (Python, Jython, IronPython) and they sometimes even change between versions. It would be a bit safer to change `IndexError: list index out of range` to `IndexError: *` in the verified message.

Also, please use four spaces as the separator. We have lot of old tests suites that still use two spaces and for consistency reasons using two is fine on them. This suite uses the nowadays recommended four-space-separator, though.